doc/classes/Curve3D.xml

```diff
@@ -137,6 +137,7 @@
 			<param index="2" name="apply_tilt" type="bool" default="false" />
 			<description>
 				Returns a [Transform3D] with [code]origin[/code] as point position, [code]basis.x[/code] as sideway vector, [code]basis.y[/code] as up vector, [code]basis.z[/code] as forward vector. When the curve length is 0, there is no reasonable way to calculate the rotation, all vectors aligned with global space axes. See also [method sample_baked].
+				Note that in the GDScript implementation, offset is expected to be normalized, that is, between zero and one. This diverges slightly from the C++ implementation, which expects an unnormalized offset.
```

Comment:
This is false, offset is in length along the curve, not normalized

Reply:
Yes, I just showed up to rescind this as I'm getting conflicting results in my own analysis code. Apologies, I'm going to work on that this morning and figure out where the issue is.

Reply:
I don't think there's any issues in the implementation, but in your testing, but in any case if there are any issues this PR is not the solution

Reply:
Perhaps I'm simply not seeing something, but my basis seems to require `progress_ratio`, while my origin seems to want `progress`, in spite of them being from the same function.

I'm ultimately attempting to align a skeleton's bones, programmatically, along a curve. Here's the code that's doing questionable things, if I may discuss it with an appropriate amount of humility:

```
func _process(delta):
	follow.progress_ratio += delta * speed
	
	#update armature
	
	#This aligns to the up vector for the curve, but we want the forward vector.
	var sample := path.curve.sample_baked_with_rotation(follow.progress_ratio).basis
	#var sample := path.curve.sample_baked_with_rotation(follow.progress).basis
	
	skeleton.set_bone_pose_rotation(0, 
		Quaternion(Vector3.RIGHT, -PI/2.0) * Quaternion(sample) #Quaternion rotation gives us forward
		)
	print(follow.progress, " vs ", path.curve.get_baked_length())
	var dist = skeleton.get_bone_global_pose(1).origin.length()
	var dist_normalized = dist / path.curve.get_baked_length()
	
	print(dist_normalized)
	#print(dist)
	
	print("T: ", follow.progress_ratio + dist_normalized)
	aim.global_position = path.curve.sample_baked_with_rotation(follow.progress + dist).origin

```

You'll note that the initial attempt to get a Transform3D used `progress` instead of `progress_ratio`, but the behavior of the bone is entirely wrong this way. When, as an experiment, I set it to `progress_ratio`, it worked out perfectly.

However, when I attempt to set the position of aim (a MeshInstance3D blue sphere) to a progressed point along the curve, it requires `progress` to function properly.

Is it possible that `_sample_posture`, which the basis is retrieved with, is expecting a normalized coordinate instead, and is returning a value on the basis of that?

Reply:
I think you're doing something wrong here, otherwise `PathFollow3D` wouldn't work as it just uses non-nirmalized offset, I think there's instead something wrong with how you use this value, or you are making incorrect assumptions about the orientation

Reply:
According to experience and the documentation, PathFollow3D supplies both progress, for rough progress in world units along its path, and progress_ratio, which is the normalized progress along it (the value usually referred to as 't' in a Bezier curve's equation).

To test this, I used a basic curve with a twist of 6π at one of its control points. When I attempt with progress, the behavior here is very strange:

![image](https://github.com/godotengine/godot/assets/7585255/d9765ba6-aec6-4d3a-82ef-e296792d9e83)

But, when I set it to progress_ratio, it behaves as expected, all the way to the end:

![image](https://github.com/godotengine/godot/assets/7585255/80dd84a9-1961-41d2-b94b-b6c5f5f1cf63)

(Note that the camera is a child of the pointy mesh here, so orientation will be slightly different; but the mesh points along the curve, as intended, only when I'm using `progress_ratio` instead of `progress`.)

On the other hand, my aim ball, which I'm using to determine whether I'm getting the appropriate position after calculating an offset, is locked close to the first 0-1 interval of the curve when I use `progress_ratio`, but behaves fine when I use `progress`.
![image](https://github.com/godotengine/godot/assets/7585255/d3671796-97c6-4b48-bcc7-5d77cb356700)
![image](https://github.com/godotengine/godot/assets/7585255/284390a8-808b-4a7d-9f5e-0b0001a25606)

This seems very much to be two different results. That leads me to believe that the C++ _sample_posture is perhaps being called with the wrong value? Are we sure that it does not expect a normalized value here?

Reply:
It does not use the ratio one though, and it doesn't call any normalized value, check the code yourself if you don't trust me 🙂, but I know it doesn't use the normalized one, simple as that

Reply:
Oh don't worry, I'm hardly taking any of this personally; but my code is performing in an unexpected way while using a few nodes which have some spotty documentation. It seems like a worthy use of my time to figure out why it's going wrong, and try and fix that.

I'm seeing that in the C++, Transform3D is constructed from two values, an origin point and a basis, sequentially. The basis is gotten with this line:

```Basis frame = _sample_posture(interval, p_apply_tilt);```

`interval` is used for the origin, too.  I'm not entirely sure what a Curve3D::Interval is yet, as it's simply returned from `_find_interval(p_offset)` and I haven't had enough time to review it, but the GDScript I showed you is pretty much all I've written for this project. I may comment it all out and try again from the top later, but it really seems to me that I should be able to call `sample_baked_with_rotation` with a single progress value and get consistent results.

To benefit this discussion, here's the full script for this project. If you think I need to share the full project then I can, it's no big deal; but I really feel like the issue is right in front of us here.

[extends Node3D

@export var speed : float = 0.1

@onready var path : Path3D = $"../.."
@onready var follow : PathFollow3D = $".."
@onready var skeleton : Skeleton3D = $"Armature/Skeleton3D"
@onready var aim : MeshInstance3D = $Aim

# Called when the node enters the scene tree for the first time.
func _ready():
	pass # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	follow.progress_ratio += delta * speed
	
	#update armature
	
	#This aligns to the up vector for the curve, but we want the forward vector.
	var sample := path.curve.sample_baked_with_rotation(follow.progress_ratio).basis
	#var sample := path.curve.sample_baked_with_rotation(follow.progress).basis
	
	# Documentation bug discovered; C++ expects unnormalized curve progress, while GDScript expects
	# normalized progress.
	
	skeleton.set_bone_pose_rotation(0, 
		Quaternion(Vector3.RIGHT, -PI/2.0) * Quaternion(sample) #Quaternion rotation gives us forward
		)
	print(follow.progress, " vs ", path.curve.get_baked_length())
	var dist = skeleton.get_bone_global_pose(1).origin.length()
	var dist_normalized = dist / path.curve.get_baked_length()
	
	print(dist_normalized)
	#print(dist)
	
	print("T: ", follow.progress_ratio + dist_normalized)
	aim.global_position = path.curve.sample_baked_with_rotation(follow.progress + dist).origin
	
](url)

Reply:
But all this aside, what ever this is caused by, the following fact remains:
The statement you've added here is false, and the method *only* takes a value that's in the range of the baked length

I'd suggest taking this question to the other community channels, but this change isn't valid

If you find an actual bug here after doing this please open a bug report, however I'll go ahead and close this as this is not a correct change

If you figure out that this is an issue with how to describe the return value further we can add that to the documentation

But the underlying assumption of this PR is wrong and the fix is also wrong

Reply:
One issue that I'm seeing is that I think PathFollow3D is possibly applying its own rotation here too... man it would be such a relief if this was just something I messed up.

Reply:
I'd suggest following the code in `PathFollow`, as it evidently works 🙂

Reply:
I'm not sure of that, but I've gotten my idea to work regardless using inverse kinematics and two PathFollow3D nodes (one for the worm, one for the target). I'm a little concerned by SkeletonIK3D being deprecated at the moment, but as far as I can tell nothing explicit has replaced it, and aside from a few workflow issues it's working well.

Likely my brain just went on strike at the end of December...

At this point I'm more than content to leave this issue closed, but thank you for your concern!